server/resp/src/main/java/org/infinispan/server/resp/Intrinsics.java

```diff
@@ -62,32 +78,40 @@ public static String bulkString(ByteBuf buf, Resp2LongProcessor longProcessor) {
       }
       // If we didn't read any bytes then the number wasn't parsed properly
       if (pos == buf.readerIndex()) {
-         return null;
+         return -1;
       }
       int size = (int) longSize;
+      if (size < 0) {
+         throw new IllegalArgumentException("Number cannot be negative");
+      }
       if (buf.readableBytes() < size + TERMINATOR_LENGTH) {
          buf.resetReaderIndex();
+         return -1;
+      }
+      return size;
+   }
+
+   public static String bulkString(ByteBuf buf, Resp2LongProcessor longProcessor) {
+      int size = readSizeAndCheckRemainder(buf, longProcessor);
+      if (size == -1) {
          return null;
       }
       String stringValue = buf.toString(buf.readerIndex(), size, StandardCharsets.US_ASCII);
       buf.skipBytes(size + TERMINATOR_LENGTH);
       return stringValue;
    }
 
-   public static byte[] bulkArray(ByteBuf buf, Resp2LongProcessor longProcessor) {
-      buf.markReaderIndex();
-      int pos = buf.readerIndex();
-      long longSize = readNumber(buf, longProcessor);
-      if (longSize > Integer.MAX_VALUE) {
-         throw new IllegalArgumentException("Bytes cannot be longer than " + Integer.MAX_VALUE);
-      }
-      // If we didn't read any bytes then the number wasn't parsed properly
-      if (pos == buf.readerIndex()) {
+   public static RespCommand bulkCommand(ByteBuf buf, Resp2LongProcessor longProcessor) {
+      int size = readSizeAndCheckRemainder(buf, longProcessor);
+      if (size == -1) {
```

Comment:
Note this uses -1 as a special place holder as the readSizeAndCheckRemainder method verifies the number is positive or 0 already. So -1 can only mean it wasn't read.